tests/bound.rs

```diff
@@ -0,0 +1,59 @@
+#![cfg(feature = "derive")]
+
+use arbitrary::{Arbitrary, Unstructured};
+
+fn arbitrary_from<'a, T: Arbitrary<'a>>(input: &'a [u8]) -> T {
+    let mut buf = Unstructured::new(input);
+    T::arbitrary(&mut buf).expect("can create arbitrary instance OK")
+}
+
+/// This wrapper trait *implies* `Arbitrary`, but the compiler isn't smart enough to work that out
+/// so when using this wrapper we *must* opt-out of the auto-generated `T: Arbitrary` bounds.
+pub trait WrapperTrait: for<'a> Arbitrary<'a> {}
+
+impl WrapperTrait for u32 {}
+
+#[derive(Arbitrary)]
+#[arbitrary(bound = "T: WrapperTrait")]
+struct GenericSingleBound<T: WrapperTrait> {
+    t: T,
+}
+
+#[test]
+fn single_bound() {
+    let v: GenericSingleBound<u32> = arbitrary_from(&[0, 0, 0, 0]);
+    assert_eq!(v.t, 0);
+}
+
+#[derive(Arbitrary)]
+#[arbitrary(bound = "T: WrapperTrait, U: WrapperTrait")]
+struct GenericMultipleBounds<T: WrapperTrait, U: WrapperTrait> {
```

Comment:
Can we also make these forms work, in addition to the above:

```rust
#[derive(Arbitary)]
#[arbitrary(bound = "T: WrapperTrait")]
#[arbitrary(bound = "U: WrapperTrait")]
struct Foo<T, U> {
    // ...
}

#[derive(Arbitary)]
#[arbitrary(
    bound = "T: WrapperTrait",
    bound = "U: WrapperTrait",
)]
struct Bar<T, U> {
    // ...
}
```

(and test all versions here)

Reply:
Good idea, implemented in 2e0ee6726c46056377eb8848325ca0e898fe2be4. As a natural consequence this also ended up supporting the empty attribute (`arbitrary()`). I was kind of thinking that would be useful anyway.